src/viur/core/utils/json.py

```diff
@@ -0,0 +1,70 @@
+import base64
+import datetime
+import json
+import pytz
+import typing as t
+from viur.core import db
+
+
+class ViURJsonEncoder(json.JSONEncoder):
+    """
+    Adds support for db.Key, db.Entity, datetime, bytes and and converts the provided obj
+    into a special dict with JSON-serializable values.
+    """
+    def default(self, obj: t.Any) -> t.Any:
+        if isinstance(obj, db.Key):
+            return {".__key__": db.encodeKey(obj)}
+        elif isinstance(obj, datetime.datetime):
+            return {".__datetime__": obj.astimezone(pytz.UTC).isoformat()}
+        elif isinstance(obj, bytes):
+            return {".__bytes__": base64.b64encode(obj).decode("ASCII")}
+        elif isinstance(obj, db.Entity):
+            # TODO: Handle SkeletonInstance as well?
+            return {
+                ".__entity__": self.default(dict(obj)),
+                ".__key__": db.encodeKey(obj.key) if obj.key else None
+            }
+        elif isinstance(obj, dict):
+            return {self.default(key): self.default(value) for key, value in obj.items()}
+        elif isinstance(obj, (list, tuple, set)):
```

Comment:
Maybe we should memorize the type. That's helpful for task chunks which want to use `set`s.

Reply:
see 72203f7149146e55065ef032685f35c6f21759b7